docs/github-merge-strategy.md

```diff
@@ -6,6 +6,43 @@ This document describes the merge strategy used for Pull Requests within all rep
 
 Every Pull Request that is made to an InstructLab repository should meet the below requirements - certain repositories such as [taxonomy](https://github.com/instruct-lab/taxonomy) may have additional requirements.
 
+### Commit Messages and PR Description
+
+PRs should be formatted as a series of logical commits that implement the
+overall change proposed in the PR. Each commit should represent a logical step
+in the progression of the change. The commit message for each should clearly
+explain that step of the progression.
+
+Here are a few examples of PRs that contained a series of commits:
+
+- <https://github.com/instructlab/instructlab-bot/pull/125/commits>
+- <https://github.com/instructlab/instructlab/pull/994/commits>
+- <https://github.com/instructlab/instructlab/pull/951/commits>
+
+Here are some external resources that provide guidance on writing good commit messages:
+
+- <https://www.berrange.com/tags/commit-message/>
+- <https://docs.kernel.org/process/submitting-patches.html#separate-your-changes>
+- <https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#smaller-is-better-small-commits-small-pull-requests>
+
+While not a requirement right now, we may want to consider following a spec like
+[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) in the
+future. It's worth a look for inspiration, at least.
+
+When your PR contains multiple commits, it is helpful to reuse the work you put
+into crafting high-quality commit messages. Here is a script you can use to generate
+a summary of the commits in your PR. It starts with a list of the commits one
+line at a time and then follows that with the full commit messages. The format works
+for pasting directly into GitHub.
```

Comment:
The GitHub CLI tools are also good for this. `hub pull-request` starts an editor with a template for creating the PR message, including all of the commit messages for commits in the branch. I'm sure the more modern `gh` does something similar, and I'd be surprised if there wasn't a VSCode somewhere that did something like it.

Reply:
I would prefer if the PR description contained a hand crafted description of the overall fix. This way the user pushing the PR has to think about the overall fix or feature being pushed. It provides a good overview to the reader of the PR afterwards.

It should contain breakdown of the different parts if helpful or can be read in the commits.